src/metabase/automagic_dashboards/populate.clj

```diff
@@ -274,6 +275,22 @@
 
 (def ^:private ^:const ^Long max-filters 4)
 
+(defn ordered-group-by-seq
+  "A seq from a group-by in a particular order. If you don't need the map itself, just to get the key value pairs in a
+  particular order. Clojure's `sorted-map-by` doesn't handle distinct keys with the same score. So this just iterates
+  over the groupby in a reasonable order."
+  [f key-order coll]
+  (letfn [(access [ks grouped]
+            (if (seq ks)
+              (let [k (first ks)]
+                (lazy-seq
+                 (if-let [x (find grouped k)]
+                   (cons x (access (next ks) (dissoc grouped k)))
+                   (access (next ks) grouped))))
+              (seq grouped)))]
+    (let [g (group-by f coll)]
+      (access key-order g))))
```

Comment:
This is to help making the groups in the dashboard. Cards each have a group. It used to use `(partition-by :group)` on the cards, and for each grouping, add a text card above the cards of the group.

But `(partition-by :group)` doesn't work well if the groups are not contiguous

```clojure
(partition-by :group [{:group 1 :x 1} {:group 2 :x 1} {:group 1 :x 2}])
-> (({:group 1, :x 1}) ({:group 2, :x 1}) ({:group 1, :x 2}))
```

So we get the key-order of the groups by their score and then pull them from the `group-by` in that order. This ensures all groups are contiguous and in the order of their score.